docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+
+With both configs in place, you can now run webpack.
+
+First you need a basic block.js to build. Create `block.js`
+
+```js
+const { registerBlockType } = wp.blocks;
+
+registerBlockType( 'myguten/test-block', {
+	title: 'Basic Example',
+	icon: 'smiley',
+	category: 'layout',
+	edit() {
+		return <div>Hola mundo!</div>;
+	},
+	save() {
+		return <div>Hola mundo!</div>;
+	},
+} );
+```
+
+To configure npm to run a script, you use the scripts section in `package.json` webpack:
+```json
+  "scripts": {
+    "build": "webpack"
+  },
+```
+
+You can then run the build using: `npm run build`
+
+Afterwards, you will see the build file created at: `block.build.js`
+
+## Finishing Touches
+
+### Development Mode
+
+The basics are in place to build. You might of noticed the webpack.config.js set a default mode to "development". Webpack has the ability to run in a "production" mode which shrinks the code down, but makes it difficult to read.
+
+The mode is setup so it can be configured using environment variables, this is easiest to add in the scripts section of `package.json`.
+
+Update scripts section to:
+
+```json
+  "scripts": {
+    "dev": "webpack --watch",
+    "build": "cross-env NODE_ENV=production webpack"
+  },
+```
+
+This sets the environment variables, but different environment handle setting in different ways, so using the `cross-env` helper module handles this. Be sure to install using: `npm install --save cross-env`
+
+Additionally, webpack has a `--watch` flag that will keep the process running, watching for any changes to the `block.js` file and update as changes occur. This is useful during development, you start the build running using `npm run dev` in a terminal. You can then edit away in your text editor and each save, webpack will automatically build. You can then use the familiar edit-save,-reload development process.
+
+**Note:** keep an eye on your terminal for any errors. If you make a typo, or syntax error, the build will fail and the error will be in the terminal.
+
+### Babel Browser TargetingT
+
+Babel has the ability to build JavaScript using rules that target certain browsers and versions. By setting a reasonable set of modern browsers allows Babel to optimize the JavaScript it generates.
+
+WordPress has a preset default you can use to target the minimum supported browsers by WordPress.
+
+Install the module using: `npm install --save @wordpress/babel-preset-default`
+
+You then update `.babelrc` adding a "presets" section:
+
+```
+{
+	"presets": [ "@wordpress/babel-preset-default" ],
+	"plugins": [
+		[ "transform-react-jsx", {
+			"pragma": "wp.element.createElement"
+		} ]
+	]
+}
+```
+
+### Source Control
+
+You should exclude `node_modules/` from your source control, the files can be recreated any time by running `npm install`.
```

Comment:
```suggestion
Because a typical `node_modules` folder will contain thousands of files that change with every software update, you should exclude `node_modules/` from your source control. If you ever start from a fresh clone, simply run `npm install` in the same folder your `package.json` is located to pull your required packages.
```